sr_robot_launch/scripts/sr_ur_arm_unlock

```diff
@@ -31,11 +32,34 @@ class SrUrUnlock():
         if 0 == len(self._arms):
             rospy.logerr("No arms detected, shutting down %s", rospy.get_name())
             rospy.signal_shutdown("No arms detected")
-        rospy.Subscriber("/sr_arm/release", Bool, self.unlock_subscriber)
+        rospy.Subscriber("/sr_arm/release_or_brake", Bool, self.release_or_brake_arm_cb)
+        rospy.wait_for_service('controller_manager/list_controllers', 5.0)
 
-    def unlock_subscriber(self, message):
+    def release_or_brake_arm_cb(self, message):
         if message.data:
-            self.release_arm()
+            if self.check_arms_needs_starting():
+                self.release_arm()
+            else:
+                for arm in self._arms:
+                    if self.is_arm_controller_running(arm):
+                        self.brake_arm(arm)
+                    else:
+                        rospy.loginfo("Controller is stopped, resending robot program to arm: %s", arm)
+                        self.call_arm_service(arm, "resend_robot_program", Trigger, dashboard=False)
+
+    def is_arm_controller_running(self, arm):
+        controller_is_running = False
+        try:
+            list_controllers = rospy.ServiceProxy('controller_manager/list_controllers', ListControllers)
+            loaded_controller = list_controllers()
+        except rospy.ServiceException, e:
+            rospy.logerr("Could not get controllers list: %s", e)
+
+        for controller in loaded_controller.controller:
+            if arm + "_trajectory_controller" in controller.name:
+                if controller.state == "running":
+                    controller_is_running = True
+                    return controller_is_running
```

Comment:
Given method name `is_arm_controller_running` it kind of feels it should always return boolean value but now it will return either `True` or `None`. Also if `ServiceException` is thrown wouldn't `loaded_controller` be undefined so should not be used in the loop? Maybe better to have something like:
```
        try:
            list_controllers = rospy.ServiceProxy('controller_manager/list_controllers', ListControllers)
            loaded_controller = list_controllers()
            for controller in loaded_controller.controller:
                if arm + "_trajectory_controller" in controller.name:
                    if controller.state == "running":
                        return True
        except rospy.ServiceException, e:
            rospy.logerr("Could not get controllers list: %s", e)
        return False
```

Reply:
wow, really didn't put enough attention here, thanks for review. addressed in https://github.com/shadow-robot/sr_interface/pull/597/commits/cfb498354aad156c2652b626e9d80cc49ec0fb91